app/Application.php

```diff
@@ -0,0 +1,227 @@
+<?php
+
+class Application
+{
+
+    protected $historyRepository;
+
+    public function __construct()
+    {
+
+        $this->historyRepository = new HistoryRepository();
+    }
+
+    public function run()
+    {
+        $isRunning = true;
+        info_box('', 'Welcome to the calculator app!', '', 'Print "help" to learn more about the app.', '');
+        while ($isRunning = true) {
+
+            $command = choose('Enter command: ', AVAILABLE_COMMANDS);
+
+            if (in_array($command, SYSTEM_COMMANDS)) {
+                $this->executeSystemCommand($command, $this->historyRepository);
+
+                continue;
+            }
+
+            $argument1 = $this->readOperand('Enter fist number: ', $command);
+
+            $argument2 = ($command !== '^' && $command !== 'sr' )
+                ? $this->readOperand('Enter second number: ', $command, true)
+                : $this->readOperand('Enter exponent: ', $command);
+
+            $result = $this->calculate($argument1, $command, $argument2);
+
+            info('Result: ' . $result);
+            write_symbol_line(25, '=');
+
+            $this->historyRepository->create($argument1, $argument2, $command, $result);
```

Comment:
date should be calculated inside the business logic code, not in the repository

```suggestion
            $this->historyRepository->create(date('d-m-Y'), $argument1, $argument2, $command, $result);
```